more_itertools/more.py

```diff
@@ -2274,11 +2273,10 @@ def __reduce__(self):
         return numeric_range, (self._start, self._stop, self._step)
 
     def __repr__(self):
+        cls_name = self.__class__.__name__
         if self._step == 1:
-            return f"numeric_range({self._start!r}, {self._stop!r})"
-        return (
-            f"numeric_range({self._start!r}, {self._stop!r}, {self._step!r})"
-        )
+            return f"{cls_name}({self._start!r}, {self._stop!r})"
+        return f"{cls_name}({self._start!r}, {self._stop!r}, {self._step!r})"
```

Comment:
Drawn from the #934 .

Reply:
The `type(self)`.__name__` was pointed out [here](https://github.com/more-itertools/more-itertools/pull/948#discussion_r1921606445), duly noted.